app/services/foreman/parameters/caster.rb

```diff
@@ -110,10 +110,14 @@ def cast_yaml
       end
 
       def load_yaml_or_json
+        self.class.load_yaml_or_json(value)
+      end
+
+      def self.load_yaml_or_json(value)
         return value unless value.is_a? String
         begin
           JSON.load value
-        rescue
+        rescue Psych::SyntaxError, JSON::ParserError
```

Comment:
Rescuing the Psych error here doesn't make sense, it has to be around the YAML.load, surely?  I still see the problem on this branch, under Ruby 1.9.3 emanating from the YAML.load line.